project/frontend/src/components/ResultsMap.js

```diff
@@ -58,34 +59,46 @@ function MapContainer(props) {
     );
   };
 
+  /** Creates all the markers that are going to be displayed
+   *  on the screen. It creates the marker corresponding
+   *  to the restaurant at currentCardIndex, the 3 markers
+   *  before that and the 3 markers after that if they are
+   *  valid indicies in the restaurants list.
+   */
   const createMarkers = () => {
-    if (!restaurants) {
-      return;
+    if (restaurants.length === 0) {
+      return null;
     }
-    let markers = [];
-    const numOfMarkers = Math.min(restaurants.length, 4);
-    for (let i = 0; i < numOfMarkers; i++) {
-      const coords = restaurants[i].key.latLngCoords;
-      const numInList = (i + 1).toString();
-      markers.push(
-        <MarkerIcon
-          lat={coords.lat}
-          lng={coords.lng}
-          id={i}
-          aria-label={'Your #' + numInList + ' Match'}
-        />
-      );
+    const markers = [];
+    const numOfMarkers = Math.min(restaurants.length, 3);
+    for (let delta = -3; delta <= numOfMarkers; delta++) {
```

Comment:
super nit: so number of markers is technically 7 here (-3 --> 3) so the naming is a bit misleading , wdyt of 
const numAdjacentMarkers = 3; and then just having the loop (delta = -numAdjacentMarkers; delta <= numAdjacentMarkers, delta++) ? 
I think we can at this point get rid of the min(restaurants.length, 3) check as now isValidIndex should handle that and if length < 3 then we end up showing all of them 

oh and also on the header comment (line 64-67) replace "3" with "numAdjacentMarkers" -- comments can fall out of sync with the actual values used so might as well leave it variable:)

Reply:
this is great! I was thinking numOfMarkers didn't make sense here anymore thank you!